pkg/didcomm/protocol/didexchange/states.go

```diff
@@ -412,3 +427,47 @@ func prepareConnectionSignature(connection *Connection) (*ConnectionSignature, e
 		Type:       "did:sov:BzCbsNYhMrjHiqZDTUASHg;spec/signature/1.0/ed25519Sha512_single",
 		SignedData: sigData}, nil
 }
+
+func (ctx *context) handleOutboundAck(msg dispatcher.DIDCommMsg) error {
+	ack := &model.Ack{}
+	if msg.OutboundDestination == nil {
+		return fmt.Errorf("outboundDestination cannot be empty for outbound Response")
+	}
+	destination := &dispatcher.Destination{
+		RecipientKeys:   msg.OutboundDestination.RecipientKeys,
+		ServiceEndpoint: msg.OutboundDestination.ServiceEndpoint,
+		RoutingKeys:     msg.OutboundDestination.RoutingKeys,
+	}
+
+	err := json.Unmarshal(msg.Payload, ack)
+	if err != nil {
+		return err
+	}
+	//TODO:Send ver key issue-299
+	sendVerKey := ""
+	return ctx.outboundDispatcher.Send(ack, sendVerKey, destination)
+}
+func (ctx *context) handleInboundResponse(response *Response) error {
+	ack := &model.Ack{
+		Type:   ConnectionAck,
+		ID:     uuid.New().String(),
+		Status: ackStatusOK,
+		Thread: &decorator.Thread{
+			ID: response.ID,
+		},
+	}
+	connBytes, err := base64.StdEncoding.DecodeString(response.ConnectionSignature.SignedData)
```

Comment:
This should be base64URL encoded - see [examples](https://github.com/hyperledger/aries-rfcs/tree/master/features/0023-did-exchange#example-1)

Reply:
changed